Matchismo/ViewController.m

```diff
@@ -14,15 +14,19 @@ @interface ViewController ()
 @property (weak, nonatomic) IBOutlet UILabel *scoreLabel;
 @property (nonatomic, strong) Deck *deck;
 @property (nonatomic, strong) CardMatchingGame *game;
+@property (weak, nonatomic) IBOutlet UISegmentedControl *gameModeSegmentedControl;
+@property (weak, nonatomic) IBOutlet UIButton *restartButton;
+@property (weak, nonatomic) IBOutlet UILabel *userActionsLabel;
 @end
 
 @implementation ViewController
 
 - (CardMatchingGame *)game
 {
-    if (_game == nil)
-    {
-        _game = [[CardMatchingGame alloc] initWithCardCounts:[self.cardButtons count] usingDeck:[self createDeck]];
+    if (_game == nil) {
+        _game = [[CardMatchingGame alloc] initWithCardCounts:[self.cardButtons count] usingDeck:[self createDeck] withMode:self.gameModeSegmentedControl.selectedSegmentIndex + 1];
+        self.gameModeSegmentedControl.enabled = NO;
+        self.restartButton.enabled = YES;
```

Comment:
does it make sense to do these actions in `updateUI` instead?

Reply:
in `updateUI` it will be updated every time I click on the button which is not good in my opinion

Reply:
Can you elaborate any downsides?